src/DDTrace/OpenTelemetry/Span.php

```diff
@@ -0,0 +1,409 @@
+     * End users should use a {@see API\TracerInterface} in order to create spans.
+     *
+     * @param non-empty-string $name
+     * @psalm-param API\SpanKind::KIND_* $kind
+     * @param list<LinkInterface> $links
+     *
+     * @internal
+     * @psalm-internal OpenTelemetry
+     */
+    public static function startSpan(
+        SpanData $span,
+        API\SpanContextInterface $context,
+        InstrumentationScopeInterface $instrumentationScope,
+        int $kind,
+        API\SpanInterface $parentSpan,
+        ContextInterface $parentContext,
+        SpanProcessorInterface $spanProcessor,
+        ResourceInfo $resource,
+        AttributesBuilderInterface $attributesBuilder,
+        array $links,
+        int $totalRecordedLinks,
+        float $startEpochNanos = 0,
+        bool $isRemapped = true // Answers the question "Was the span created using the OTel API?"
+    ): self {
+        $attributes = $attributesBuilder->build()->toArray();
+        self::_setAttributes($span, $attributes); // Counts towards the attribute limit since it's derived from the builder
+
+        $resourceAttributes = $resource->getAttributes()->toArray();
+        self::_setAttributes($span, $resourceAttributes); // Doesn't count towards the attribute limit, and shouldn't - special case
+
+        if ($isRemapped || empty($span->name)) {
+            // Since the span was created using the OTel API, it doesn't have an operation name*
+            // *: This is a bit false. For instance, a span created using the OTel API and DD_TRACE_GENERATE_ROOT_SPAN=0
+            // under a cli (e.g., phpunit) process would have a default operation name set (e.g., phpunit). This is
+            // done in serializer.c:ddtrace_set_root_span_properties (as of v0.92.0)
+            $span->name = Convention::defaultOperationName($span);
+        }
+
+        $OTelSpan = new self(
+            $span,
+            $context,
+            $instrumentationScope,
+            $kind,
+            $parentSpan->getContext(),
+            $spanProcessor,
+            $resource,
+            $attributesBuilder
+        );
+
+        ObjectKVStore::put($span, 'otel_span', $OTelSpan);
+
+        // Call onStart here to ensure the span is fully initialized.
+        $spanProcessor->onStart($OTelSpan, $parentContext);
+
+        return $OTelSpan;
+    }
+
+    public function getName(): string
+    {
+        return $this->span->resource ?: $this->span->name;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getContext(): SpanContextInterface
+    {
+        return $this->context;
+    }
+
+    public function getParentContext(): API\SpanContextInterface
+    {
+        return $this->parentSpanContext;
+    }
+
+    public function getInstrumentationScope(): InstrumentationScopeInterface
+    {
+        return $this->instrumentationScope;
+    }
+
+    public function hasEnded(): bool
+    {
+        return $this->hasEnded;
```

Comment:
Is this intentionally just checking for hasEnded and not getDuration() != null, which is only triggered when Context::current is called?

```
start_span();
$otelSpan = Context::current();
close_span();
var_dump($otelSpan->hasEnded()); // false?
```

Reply:
Speaking about `toSpanData`, that just made me realized that... the attributes may be wrong if someone has the wonderful idea of _unsetting_ attributes from the DD-span...

Reply:
Especially if you unset